src/main/java/cn/danielw/fop/PoolConfig.java

```diff
@@ -12,6 +12,7 @@
     private int partitionSize = 4;
     private int scavengeIntervalMilliseconds = 1000 * 60 * 2;
     private double scavengeRatio = 0.5; // avoid cleaning up all connections in the pool at the same time
+    private long shutdownWaitMilliseconds = 1000 * 30;
```

Comment:
```suggestion
    private long shutdownWaitMilliseconds = 1000L * 30;
```
SonarCloud marks this as a bug, even though statically the product can be seen to be in integer bounds